Resources/Catalog/FileCatalog.py

```diff
@@ -1,13 +1,62 @@
 """ File catalog class. This is a simple dispatcher for the file catalog plug-ins.
     It ensures that all operations are performed on the desired catalogs.
+
+    The File Catalog plug-ins are supposed to implement all or some of the
+    methods described below. In all the methods the LFNs argument can have one
+    of the following forms:
+
+    - string: just a single LFN itself
+    - list: a list of LFN strings
+    - dictionary: the keys are LFN strings, the values are LFN specific parameters
+                  needed by the method
+
+    All the methods are returning a standard DIRAC bulk result structure. If the
+    call is successful, the Successful and Failed dictionaries have LFNs as keys
+    and specific results of operation as values.
+
+    The FileCatalog methods:
+
+    exists( LFNs ) - check the existence of the LFNs
+
+    addFile( LFNs )
+    removeFile
+    isFile( LFNs )
+    setFileStatus
+    getFileMetadata
+    getFileSize
+
+    createDirectory
+    removeDirectory
+    isDirectory
+    setDirectoryStatus
+    listDirectory
+    getDirectoryMetadata
+    getDirectorySize
+    getDirectoryContents
+
+    addReplica
+    removeReplica
+    setReplicaStatus
+    setReplicaHost
+    setReplicaProblematic
+    getReplicas
+    getReplicaStatus
+    getDirectoryReplicas
+
+    createLink
+    removeLink
+    isLink
+    readLink
+
+
 """
 
-import types, re
+import re
 
 from DIRAC  import gLogger, gConfig, S_OK, S_ERROR
 from DIRAC.ConfigurationSystem.Client.Helpers.Operations    import Operations
 from DIRAC.Core.Security.ProxyInfo                          import getVOfromProxyGroup
-from DIRAC.Resources.Utilities                              import checkArgumentFormat
+from DIRAC.DataManagementSystem.DB.FileCatalogComponents.Utilities import checkArgumentFormat
```

Comment:
We have essentially the same thing in 2 different places. This one is more complete. I am not sure that this is the best place to keep this utility but it should rather be in the DataManagement